### PR TITLE
Refactor elastic correction workflow

### PR DESCRIPTION
**Pull request by jmborr**
## Description of work:
Refactor the EQSANS elastic reference correction workflow by extracting reusable helpers for computing wavelength-dependent K factors and for writing pre/post-correction wavelength profiles, and by tightening input validation around K-factor vector shapes.
Update the main `elastic_correction()` workflow to compute K(λ) factors via the new helper and adds unit tests covering the new behaviors and validation.

**Changes:**
- Added `calculate_elastic_reference_k_factors()` (K-only helper) and `save_wavelength_dependent_profiles()` (debug/profile output helper).
- Added strict validation to reject invalid `k_vec` / `k_error_vec` shapes and lengths for both 1D and 2D normalization paths.
- Added/expanded unit tests to cover profile output and K-vector validation edge cases.

**Check all that apply:**
- [ ] ~updated documentation and checked that it looks correct in the [pull request preview](https://docs.readthedocs.com/platform/stable/pull-requests.html)~
- [x] Source added/refactored
- [x] Added unit tests
- [ ] ~Added integration tests~
- [ ] Included a manual test for the reviewer
- [ ] Verified that tests requiring the /SNS and /HFIR filesystems pass without fail

**References:**
- Links to IBM EWM items: [16511](https://ornlrse.clm.ibmcloud.com/ccm/web/projects/Neutron%20Data%20Project%20%28Change%20Management%29#action=com.ibm.team.workitem.viewWorkItem&id=16511)

## Check list for the reviewer
- [ ] best software practices
    + [ ] clearly named variables (better to be verbose in variable names)
    + [ ] code comments explaining the intent of code blocks
- [ ] All the tests are passing
- [ ] The documentation is up to date and looks correct in the [pull request preview](https://docs.readthedocs.com/platform/stable/pull-requests.html)
- [ ] code comments added when explaining intent

### Execution of tests requiring the /SNS and /HFIR filesystems
It is strongly encouraged that the reviewer runs the "manual" tests in their local machine
because these are not run by the GitLab CI. It is necessary that the remote /SNS and /HFIR filesystems
are mounted in the machine running the tests.
In the below code block, substitute `<MERGE_REQUEST_NUMBER>` for the actual merge request number`

```bash
cd /path/to/my/drtsans/
git fetch origin merge-requests/<MERGE_REQUEST_NUMBER>/head:mr<MERGE_REQUEST_NUMBER>
git switch mr<MERGE_REQUEST_NUMBER>
pixi run manual-test
```

### CODE REVIEW

File: src/drtsans/tof/eqsans/elastic_correction.py [hunk 1]
```diff
@@ -183,6 +183,107 @@ def calculate_elastic_reference_normalization(wl_vec, x_vec, ref_i_1d):
...
+        save_i1d(before_i1d, before_path)
+
+    normalized_intensity, normalized_error = normalize_intensity_1d(
+        wl_vec,
+        x_vec,
+        i_array,
+        error_array,
+        k_vec,
+        k_error_vec,
+    )
```
[the remark this answers is not in the document]
Comment (jmborr) [reply src/drtsans/tof/eqsans/elastic_correction.py:229]: Addressed in 202d6166, with the validation placed at the shared normalization boundary rather than in `save_wavelength_dependent_profiles`.

That helper is diagnostics-only and runs only when `output_wavelength_dependent_profile` and `output_dir` are both set, so validating there would have left the actual normalization paths unguarded and only fired when profile output happened to be enabled. Instead there is now a `_validate_k_vectors` helper called from `normalize_intensity_1d` (which both 1D entry points, including the profile helper, funnel through) and from `normalize_by_elastic_reference_2d`.

Two notes on the analysis:

- The 1D case worth guarding is narrower than a plain length mismatch, which already raises a broadcast `ValueError`. The silent one is a `k_vec` shaped `(number of Q or phi, 1)`, which broadcasts along the Q axis and scales each Q bin instead of each wavelength.
- The 2D case is a different failure entirely: the scaling loops are bounded by the data's wavelength count rather than by the length of the supplied factors, so an *overlong* `k_vec` has its trailing entries silently left unread. That is the one that could misalign factors with wavelengths.

The guard uses `np.ndim`/`np.size` rather than attribute access, since existing callers pass plain Python lists.

One related point for the record: single-wavelength-bin operation cannot reach this code. `bypass_correction_for_single_wavelength_bin` in `src/drtsans/tof/eqsans/correction_api.py` disables elastic correction when only one mid-bin wavelength is present, and `reduction_api.py` calls it before any correction runs.

**Review by darshdinger — APPROVED** [review]
Looks good!
